symengine/functions.cpp

```diff
@@ -2895,6 +2895,108 @@ RCP<const Basic> uppergamma(const RCP<const Basic> &s, const RCP<const Basic> &x
 }
 
 
+PolyGamma::PolyGamma(const RCP<const Basic> &n, const RCP<const Basic> &x)
+    : n_{n}, x_{x}
+{
+    SYMENGINE_ASSERT(is_canonical(n_, x_))
+}
+
+bool PolyGamma::is_canonical(const RCP<const Basic> &n, const RCP<const Basic> &x)
+{
+    if(is_a<Integer>(*n)) {
+        auto n_int = rcp_static_cast<const Integer>(n); 
+        if(n_int->is_negative()) {
+            return false;
```

Comment:
I assumed that for negative `n` `polygamma` is `complex infinity`. 
But it seems for negative `n` it can represented as some closed forms also.

I'll make corrections. Thanks